src/group_theory/group_action/support.lean

```diff
@@ -0,0 +1,53 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.set.pointwise
+
+/-!
+# Support of an element under an action action
+
+Given an action of a group `G` on a type `α`, we say that a set `s : set α ` supports an element
+`a : α` if, for all `g`, `g` fixes `a` if `g` fixes `s` pointwise.
+
+This is crucial in Fourier-Motzkin constructions.
+-/
+
+open_locale pointwise
+
+variables {G H α β : Type*}
+
+namespace mul_action
+section has_smul
+variables (G) [has_smul G α] [has_smul G β]
+
+/-- A set `s` supports `b` if `g • b = b` whenever `g • a = a` for all `a ∈ s`. -/
+@[to_additive "A set `s` supports `b` if `g +ᵥ b = b` whenever `g +ᵥ a = a` for all `a ∈ s`."]
+def supports (s : set α) (b : β) := ∀ g : G, (∀ ⦃a⦄, a ∈ s → g • a = a) → g • b = b
+
+variables {s t : set α} {a : α} {b : β}
+
+@[to_additive] lemma supports_of_mem (ha : a ∈ s) : supports G s a := λ g h, h ha
+
+variables {G}
+
+@[to_additive] lemma supports.mono (h : s ⊆ t) (hs : supports G s b) : supports G t b :=
```

Comment:
I suppose the `@[mono]` attribute won't do anything useful here.

Reply:
Yeah, I'm afraid it won't recognise this as a relation.